pkg/app/app.go

```diff
@@ -794,9 +794,10 @@ func (a *App) visitStates(fileOrDir string, defOpts LoadOpts, converge func(*sta
 			if len(st.Helmfiles) > 0 {
 				noMatchInSubHelmfiles := true
 				for i, m := range st.Helmfiles {
+					overrideValues := append(defOpts.Environment.OverrideValues, m.Environment.OverrideValues...)
 					optsForNestedState := LoadOpts{
 						CalleePath:        filepath.Join(d, f),
-						Environment:       m.Environment,
+						Environment:       state.SubhelmfileEnvironmentSpec{OverrideValues: overrideValues},
```

Comment:
Thanks for your effort! But this is incorrect as you are now unable to specify `values` for each sub `helmfiles:` entry.

Can you just explicitly inherit values using go template?

Reply:
Ah well sorry I forgot that I already mentioned that in https://github.com/roboll/helmfile/issues/2027#issuecomment-1001465338 😅